Automattic/wp-calypso
# Gravatar: Customize page's title and favicon for the passwordless login flow

Related to 105879-Automattic/gravatar

## Proposed Changes

* Customize the page's title and favicon for Gravatar passwordless login flow in SSR
* Customize the page's title and favicon for Gravatar passwordless login flow in CSR
* Add a `faviconUrl` prop for the Head component

## Testing Instructions



* Sandbox `public-api.wordpress.com`, and then apply this diff: `arc patch D123169`
* Setup Calypso, and run `yarn start`
* (Test SSR) Log out of your account. Go to the login page of [Gravatar](http://calypso.localhost:3000/log-in/link?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3Dbfa7ea95b0a07d39e3f23e1d923ce77b35016295d986b51361d4334a6e571206%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1) and [WPJM](http://calypso.localhost:3000/log-in/link/?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1). The page's title and favicon should work as below:

| Grav | WPJM |
| - | - |
| <img width="260" alt="截圖 2023-09-26 下午9 00 24" src="https://github.com/Automattic/wp-calypso/assets/21308003/e6caf182-6cb5-4dfd-829b-cee6d2747e2e"> | <img width="259" alt="截圖 2023-09-26 下午9 00 52" src="https://github.com/Automattic/wp-calypso/assets/21308003/55b4a0b7-9bdf-4f0f-a4c1-d917357cc1b9"> |

* (Test CSR) Click the "Sign in another way" link, the password / social login page, the page's title and favicon should work  as below:

| Grav | WPJM |
| - | - |
| <img width="260" alt="截圖 2023-09-26 下午9 00 24" src="https://github.com/Automattic/wp-calypso/assets/21308003/e6caf182-6cb5-4dfd-829b-cee6d2747e2e"> | <img width="259" alt="截圖 2023-09-26 下午9 00 52" src="https://github.com/Automattic/wp-calypso/assets/21308003/55b4a0b7-9bdf-4f0f-a4c1-d917357cc1b9"> |

* Login and then confirm your login link. The page's title and favicon on the loading page should work as below and **the page should work the same**:

| Grav | WPJM |
| - | - |
| <img width="260" alt="截圖 2023-09-26 下午9 00 24" src="https://github.com/Automattic/wp-calypso/assets/21308003/e6caf182-6cb5-4dfd-829b-cee6d2747e2e"> | <img width="259" alt="截圖 2023-09-26 下午9 00 52" src="https://github.com/Automattic/wp-calypso/assets/21308003/55b4a0b7-9bdf-4f0f-a4c1-d917357cc1b9"> |

* Check [VideoPress's signup page](http://calypso.localhost:3000/start/videopress-account/user/en?variationName=videopress&flow=videopress&pageTitle=Video%20Portfolio&redirect_to=/setup/videopress/options), it should still work

* The page's title and favicon of WP.com's [login page](http://calypso.localhost:3000/log-in), [magic login page](http://calypso.localhost:3000/log-in/link), and the related login pages still work the same.

## Pre-merge Checklist
